tree/dataframe/src/RDFActionHelpers.cxx

```diff
@@ -242,6 +243,56 @@ void ValidateSnapshotOutput(const RSnapshotOptions &opts, const std::string &tre
    }
 }
 
+#ifdef R__HAS_ROOT7
+void ValidateSnapshotRNTupleOutput(const RSnapshotOptions &opts, const std::string &ntupleName,
+                                   const std::string &fileName)
+{
+   TString fileMode = opts.fMode;
+   fileMode.ToLower();
+   if (fileMode != "update")
+      return;
+
+   // output file opened in "update" mode: must check whether output RNTuple is already present in file
+   std::unique_ptr<TFile> outFile{TFile::Open(fileName.c_str(), "update")};
+   if (!outFile || outFile->IsZombie())
+      throw std::invalid_argument("Snapshot: cannot open file \"" + fileName + "\" in update mode");
+
+   auto *outNTuple = outFile->Get<ROOT::Experimental::RNTuple>(ntupleName.c_str());
+
+   if (outNTuple) {
+      if (opts.fOverwriteIfExists) {
+         outFile->Delete(ntupleName.c_str());
+         return;
+      } else {
+         const std::string msg = "Snapshot: RNTuple \"" + ntupleName + "\" already present in file \"" + fileName +
+                                 "\". If you want to delete the original ntuple and write another, please set "
+                                 "RSnapshotOptions::fOverwriteIfExists to true.";
+         throw std::invalid_argument(msg);
+      }
+   }
+
+   // Also check if there is any object other than an RNTuple with the provided ntupleName.
+   TObject *outObj = outFile->Get(ntupleName.c_str());
+
+   if (!outObj)
+      return;
+
+   // An object called ntupleName is already present in the file.
+   if (opts.fOverwriteIfExists) {
+      if (outObj->InheritsFrom("TTree")) {
+         static_cast<TTree *>(outObj)->Delete("all");
+      } else {
+         outFile->Delete(ntupleName.c_str());
```

Comment:
I think here and other `outFile->Delete` calls you might mean:
```suggestion
         outFile->Delete((ntupleName + ";*").c_str());
```
otherwise the `TTree` case is the only one deleting older cycles.

Reply:
It has been changed to:

https://github.com/enirolf/root/blob/bf7088ea7bc4a25d41c36dec2e13280c3cda61e2/tree/dataframe/src/RDFActionHelpers.cxx#L282-L287